vars/util.groovy

```diff
@@ -615,8 +622,16 @@ def withKerberos(String keytabId, Closure closure, String domain = 'REDHAT.COM')
         def currentPrincipal = sh(returnStdout: true, script: "klist | grep -i 'Default principal' | awk -F':' 'NR==1{print \$2}' ").trim()
 
         if (currentPrincipal != env.KERBEROS_PRINCIPAL) {
-            def kerberosStatus = sh(returnStatus: true, script: "kinit ${env.KERBEROS_PRINCIPAL} -kt $KEYTAB_FILE")
+            def kerberosStatus = 0
+            for (int i = 0; i < nRetries; i++) {
```

Comment:
@Ginxo @rgdoliveira do you think should I add a small sleep/delay after each run?

Reply:
@lampajr IMO I don't think it is needed.